packages/opentelemetry-resources/src/platform/node/ProcessDetectorSync.ts

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { diag } from '@opentelemetry/api';
+import { SemanticResourceAttributes } from '@opentelemetry/semantic-conventions';
+import { Resource } from '../../Resource';
+import { DetectorSync, ResourceAttributes } from '../../types';
+import { ResourceDetectionConfig } from '../../config';
+import { IResource } from '../../IResource';
+import * as os from 'os';
+
+/**
+ * ProcessDetectorSync will be used to detect the resources related current process running
+ * and being instrumented from the NodeJS Process module.
+ */
+class ProcessDetectorSync implements DetectorSync {
+  detect(_config?: ResourceDetectionConfig): IResource {
+    const attributes: ResourceAttributes = {
+      [SemanticResourceAttributes.PROCESS_PID]: process.pid,
+      [SemanticResourceAttributes.PROCESS_EXECUTABLE_NAME]: process.title,
+      [SemanticResourceAttributes.PROCESS_EXECUTABLE_PATH]: process.execPath,
+      [SemanticResourceAttributes.PROCESS_COMMAND_LINE]: process.argv.join(' '),
+      [SemanticResourceAttributes.PROCESS_COMMAND_ARGS]: process.argv,
```

Comment:
argv misses node-specific command line arguments:

```sh
$ cat test.js 
console.log('process.argv', process.argv)
console.log('process.execArgv', process.execArgv)
console.log('all arguments', [process.argv0, ...process.execArgv, ...process.argv.slice(1)])

$ node --max-old-space-size=4096 test.js arg1 arg2
process.argv [
  '/Users/daniel.dyla/.nvm/versions/node/v14.18.1/bin/node',
  '/Users/daniel.dyla/otel/opentelemetry-js-contrib/test.js',
  'arg1',
  'arg2'
]
process.execArgv [ '--max-old-space-size=4096' ]
all arguments [
  'node',
  '--max-old-space-size=4096',
  '/Users/daniel.dyla/otel/opentelemetry-js-contrib/test.js',
  'arg1',
  'arg2'
]
```

I prefer `process.argv[0]` (full path) to `process.argv0` (actual string used in command). The spec uses `argv0` in the example.

My preference would be `[process.argv[0], ...process.execArgv, ...process.argv.slice(1)]`